### ISSUE TITLE
Configure Server <udn/>

### ISSUE THREAD

**Issue by seppbiersack** [gerbera/gerbera#350]
OS: Ubuntu 18.04.1
Gerbera version 1.2.0, installed from the  Ubuntu PPA [https://launchpad.net/~stephenczetty/+archive/ubuntu/gerbera-updates](url)

An excerpt from the documentation [http://docs.gerbera.io/en/latest/config-server.html#udn](url)

udn
`<udn/>`
Required
Default: **automatically generated if the tag is empty**
Unique Device Name, according to the UPnP spec it must be consistent throughout reboots. You can fill in something yourself, but **we suggest that you leave this tag empty - it will be filled out and saved automatically after the first launch of the `server**.

_**But nothing is generated!**_

At the first launch of the server I get an error message **ERROR: Config: option not found: /server/udn**
Here is an excerpt from the file /var/log/syslog:

Sep 24 21:35:47 t430s systemd[1]: Started Gerbera Media Server.
Sep 24 21:35:47 t430s gerbera[12760]: 2018-09-24 21:35:47    INFO: Gerbera UPnP Server version 1.2.0 - http://gerbera.io/
Sep 24 21:35:47 t430s gerbera[12760]: 2018-09-24 21:35:47    INFO: ===============================================================================
Sep 24 21:35:47 t430s gerbera[12760]: 2018-09-24 21:35:47    INFO: Gerbera is free software, covered by the GNU General Public License version 2
Sep 24 21:35:47 t430s gerbera[12760]: 2018-09-24 21:35:47    INFO: Copyright 2016-2018 Gerbera Contributors.
Sep 24 21:35:47 t430s gerbera[12760]: 2018-09-24 21:35:47    INFO: Gerbera is based on MediaTomb: Copyright 2005-2010 Gena Batsyan, Sergey Bostandzhyan, Leonhard Wimmer.
Sep 24 21:35:47 t430s gerbera[12760]: 2018-09-24 21:35:47    INFO: ===============================================================================
Sep 24 21:35:47 t430s gerbera[12760]: 2018-09-24 21:35:47    INFO: **Loading configuration from: /etc/gerbera/config.xml**
Sep 24 21:35:47 t430s gerbera[12760]: 2018-09-24 21:35:47    INFO: Checking configuration...
Sep 24 21:35:47 t430s gerbera[12760]: 2018-09-24 21:35:47   **ERROR: Config: option not found: /server/udn**
Sep 24 21:35:47 t430s systemd[1]: gerbera.service: Main process exited, code=exited, status=1/FAILURE
Sep 24 21:35:47 t430s systemd[1]: gerbera.service: Failed with result 'exit-code'.
Sep 24 21:35:52 t430s systemd[1]: gerbera.service: Service hold-off time over, scheduling restart.
Sep 24 21:35:52 t430s systemd[1]: gerbera.service: Scheduled restart job, restart counter is at 1.
Sep 24 21:35:52 t430s systemd[1]: Stopped Gerbera Media Server.

**Comment by v00d00** [comment]
Good point! That documentation is outdated.

This process was revamped by @elmodaddyb, who can elaborate on it.

tl;dr run `--create-config` and it should generate you a config with a UDN.

**Comment by elmodaddyb** [comment]
@seppbiersack - Please excuse the missing documentation.  We removed runtime generation of the `<udn>` to separate invocation using the `--create-config` option to solve #196 for Gentoo.

I will include updated documentation that links to how to [generate the configuration](http://docs.gerbera.io/en/latest/config-overview.html#generating-configuration)

If `--create-config` does not work for you please post back...and will take a look.

> Thanks, Eamonn

**Comment by seppbiersack** [comment]
Some things in advance:
1. You have left the designation `<udn/>` (empty tag) in the documentation. This is confusing in my opinion. It should rather look like this: `<udn>uuid:something_uuid_like</udn>`, so that the reader would know what the tag should look like.
2. I tested the `--create-config` option. I pointed out that the 1st line of the root tag will not be generated (`<? Xml version = "1.0" encoding = "UTF-8"?>`).

And now some considerations of a general nature.
Now I understand why I did not know about the `<udn>` tag for the Mediatomb and the Gerbera versions before 1.2: earlier was this tag automatically filled by the installation routine, so that Mediatomb worked out-of-the-box and Gerbera at least let itself be started.

If you want to force the user, that he must generate the file config.xml itself, no file `/etc/gerbera/config.xml` should be created during the installation (especially with the faulty tag `<udn/>`), so the user then see through a message, that he should generate the file `config.xml`. But where should a user, who has just installed a Gerbera package via GUI of package manager and is not a Linux freak and never heard about the `/var/log/syslog`, see this message?!

Because under Ubuntu Gerbera will be automatically started by the `systemd` by default, no user will get the idea to start Gerbera from the command line and consequently he will not see any error messages.

By the way, I turn off the automatism permanently:
`sudo systemctl disable gerbera.service`
Check:
`systemctl is-enabled gerbera.service`
 and start/stop Gerbera via GUI with selfmade scripts:

```
$ cat /usr/share/applications/gerbera-server-start.desktop
[Desktop Entry]
Name=Gerbera server start
Comment=Starts Gerbera server
Exec=sudo systemctl start gerbera.service
Terminal=true
Icon=gerbera.png
Type=Application
Categories=AudioVideo;

$ cat /usr/share/applications/gerbera-server-stop.desktop
[Desktop Entry]
Name=Gerbera server stop
Comment=Stop Gerbera server
Exec=sudo systemctl stop gerbera.service
Terminal=true
Icon=gerbera.png
Type=Application
Categories=AudioVideo;
```

No, `config.xml` should be automatically generated during installation, as was the case with Mediatomb. You should not scare the user.

**Comment by elmodaddyb** [comment]
@seppbiersack - These are all great points.  Thanks for taking the time to post the feedback.

> I will update the documentation to clarify `<udn>uuid:[generated-uuid]</udn>`

As for the generation of the `config.xml`, would need to investigate and see if this is added as part of the Package Manager **packaging** phase, since I do not see it as part of `CMakeLists.txt`

> Thanks again for all the help.
